### PR TITLE
Full Hashicorp Vault integration

### PR DESCRIPTION
**Pull request by relistan**
This builds on the existing Vault integration by moving all the login and auth code into the executor itself in Go, rather than relying on the external scripting to provider a valid token. Currently supports userpass auth, which is what has been supported in the shell script.

### CODE REVIEW

**Comment by relistan** [comment]
cc @dselans

File: vault/vault.go [hunk 1]
```diff
@@ -38,7 +54,16 @@ func NewDefaultVault() EnvVault {
 		log.Warnf("Unable to load Environment vars: %s", err)
 	}
 	log.Infof("Vault address '%s' ", conf.Address)
+
 	vaultClient, _ := api.NewClient(conf)
+
+	if os.Getenv("VAULT_TOKEN") == "" {
```
Comment (sbz) [inline vault/vault.go]: Just starting to look at that changes, and I have noticed, the api function [NewClient()](https://github.com/Nitro/sidecar-executor/blob/master/vendor/github.com/hashicorp/vault/api/client.go#L345) already lookup for the `VAULT_TOKEN` environment variable per [client.go#L378](https://github.com/Nitro/sidecar-executor/blob/master/vendor/github.com/hashicorp/vault/api/client.go#L378) and set the `token` parameter into `client.token` on the `client` instance. Here `vaultClient` above, I guess we can remove this `os.GetEnv` call.

**Comment by sbz** [comment]
Also, it's great you implemented the `token cache` reuse logic from `VAULT_TOKEN_FILE` directly in the executor instead the wrapper in bash script we had, great job as usual! 💯

File: vault/vault_auth.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultTokenTTL  = 86400 // 1 day
```
Comment (sbz) [inline vault/vault_auth.go]: Should we use `time.Second` unit for the default values to specify the unit is `Second`? Similar to [here](https://github.com/Nitro/sidecar/blob/master/discovery/discovery.go#L11)

```go
DefaultTokenTTL = 86400 * time.Second // 1 day
TokenGracePeriod = 3600 * time.Second // 1 hour
```

File: vault/vault.go (hunk 1 above)
Comment (relistan) [reply vault/vault.go]: Well, fair point, but we actually need to know if the env var was set or not, as the source of truth. I _could_ call `client.Token()` here instead, but then it's quite opaque where the token came from. This way you can read the code and right away see what the logic hangs on.

**Pull request merged**